web/src/reminders_overlay_ui.ts

```diff
@@ -0,0 +1,130 @@
+
+import * as browser_history from "./browser_history.ts";
+import * as message_reminder from "./message_reminder.ts";
+import type {Reminder} from "./message_reminder.ts";
+import * as messages_overlay_ui from "./messages_overlay_ui.ts";
+import * as overlays from "./overlays.ts";
+import * as timerender from "./timerender.ts";
+
+type ReminderRenderContext = Reminder & {
+    formatted_send_at_time: string;
+};
+
+export const keyboard_handling_context = {
+    get_items_ids() {
+        const reminders_ids = [];
+        const sorted_messages = sort_reminders(message_reminder.reminders_data);
+        for (const message of sorted_messages) {
+            reminders_ids.push(message.scheduled_message_id.toString());
+        }
+        return reminders_ids;
+    },
+    on_enter() {
+        // TODO: Allow editing reminder.
+        return;
+    },
+    on_delete() {
+        const focused_element_id = messages_overlay_ui.get_focused_element_id(this);
+        if (focused_element_id === undefined) {
+            return;
+        }
+        const $focused_row = messages_overlay_ui.row_with_focus(this);
+        messages_overlay_ui.focus_on_sibling_element(this);
+        // We need to have a super responsive UI feedback here, so we remove the row from the DOM manually
+        $focused_row.remove();
+        message_reminder.delete_reminder(Number.parseInt(focused_element_id, 10));
+    },
+    items_container_selector: "reminders-container",
+    items_list_selector: "reminders-list",
+    row_item_selector: "reminder-row",
+    box_item_selector: "reminder-info-box",
+    id_attribute_name: "data-reminder-id",
+};
+
+function sort_reminders(reminders: Map<number, Reminder>): Reminder[] {
+    const sorted_messages = [...reminders.values()].sort(
+        (msg1, msg2) => msg1.scheduled_delivery_timestamp - msg2.scheduled_delivery_timestamp,
+    );
+    return sorted_messages;
+}
+
+export function handle_keyboard_events(event_key: string): void {
+    messages_overlay_ui.modals_handle_events(event_key, keyboard_handling_context);
+}
+
+function format(reminders: Map<number, Reminder>): ReminderRenderContext[] {
+    const formatted_msgs = [];
+    const sorted_messages = sort_reminders(reminders);
+
+    for (const msg of sorted_messages) {
+        const time = new Date(msg.scheduled_delivery_timestamp * 1000);
+        const formatted_send_at_time = timerender.get_full_datetime(time, "time");
+        const msg_render_context = {
+            ...msg,
+            formatted_send_at_time,
+        };
+        formatted_msgs.push(msg_render_context);
+    }
+    return formatted_msgs;
+}
+
+export function launch(): void {
+    $("#reminders-overlay-container").html(render_reminders_overlay());
+    overlays.open_overlay({
+        name: "reminders",
+        $overlay: $("#reminders-overlay"),
+        on_close() {
+            browser_history.exit_overlay();
+        },
+    });
+
+    const rendered_list = render_reminder({
```

Comment:
`render_reminder_list` or something would be a better name for the template; that'd avoid the singular/plural confusion present here.

Reply:
Done